spec/solver_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'solver'
+
+describe Solver do
+  context 'To instantiate Solver object and tests its methods' do
+    it 'can output factorial values' do
+      solver = Solver.new
+      expect(solver.factorial(5)).to eq(120)
+    end
```

Comment:
- [ ] Kindly include more tests for different scenarios as contained in the guideline, such as

  - When a number is zero.
  - When a number is negative (raise an exception)

  ![factorial-1](https://user-images.githubusercontent.com/54848363/182941856-e13fa3e0-d3eb-4d38-adba-4930e813a6bb.jpg)